app/services/discourse_subscriptions/campaign.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+module DiscourseSubscriptions
+  class Campaign
+    include DiscourseSubscriptions::Stripe
+
+    def initialize
+      set_api_key # instantiates Stripe API
+    end
+
+    def refresh_data
+      product_ids = Product.all.pluck(:external_id)
+
+      # if a product id is set for the campaign, we only want to return those results.
+      # if it's blank, return them all.
+      campaign_product = SiteSetting.discourse_subscriptions_campaign_product
+      product_ids = product_ids.select { |id| id == campaign_product } if campaign_product.present?
+
+      amount = 0
+      subscriptions = get_subscription_data
+      subscriptions = filter_to_subscriptions_products(subscriptions, product_ids)
+
+      # get number of subscribers
+      SiteSetting.discourse_subscriptions_campaign_subscribers = subscriptions&.length || 0
+
+      # calculate amount raised
+      subscriptions&.map do |sub|
```

Comment:
```suggestion
      subscriptions&.each do |sub|
```
You're not getting the return value of the `map` so `each` makes more sense.